common/sampler/sampler.cpp

```diff
@@ -0,0 +1,186 @@
+#include <cinttypes>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cxxabi.h>
+#include <dlfcn.h>
+#include <impl/Kokkos_Profiling_Interface.hpp>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <tool_chain.hpp>
+#include <unordered_set>
+#include <vector>
+static Kokkos::Tools::callback_chainer next;
+using action_type = Kokkos::Tools::Experimental::ToolActions;
+action_type actions;
+static uint64_t sample_rate = 101;
+
+extern "C" void kokkosp_begin_parallel_scan(const char *name,
+                                            const uint32_t devID,
+                                            uint64_t *kID) {
+  *kID = 0;
+  static uint64_t encounter;
+  ++encounter;
+  if ((encounter % sample_rate) == 0) {
+    actions.fence(0);
```

Comment:
Making a comment for myself (David, you don't need to respond to this of course) that I think this parameter to fence function should not have to be zero.